You cannot SPECIALISE ‘HMap.insert’

With the current master and GHC 8.10.7, I am getting the following error when building Agda with `make install-bin` or `make v1-install`:

```
src/full/Agda/TypeChecking/Monad/Base.hs:1780:1: error: [-Werror]
    You cannot SPECIALISE ‘HMap.insert’
      because its definition has no INLINE/INLINABLE pragma
      (or its defining module ‘Data.HashMap.Internal.Strict’
       was compiled without -O)
     |
1780 | {-# SPECIALIZE HMap.insert :: QName -> v -> HashMap QName v -> HashMap QName v #-}
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

@AndrasKovacs could this be caused by one of your recent commits?

--- a/src/full/Agda/TypeChecking/Monad/Base.hs
+++ b/src/full/Agda/TypeChecking/Monad/Base.hs
@@ -1777,7 +1777,10 @@ type Definitions = HashMap QName Definition
 type RewriteRuleMap = HashMap QName RewriteRules
 type DisplayForms = HashMap QName [LocalDisplayForm]
 
+-- 2023-21-30, András: see issue 6927
+#if __GLASGOW_HASKELL__ >= 900
 {-# SPECIALIZE HMap.insert :: QName -> v -> HashMap QName v -> HashMap QName v #-}
+#endif
 {-# SPECIALIZE HMap.lookup :: QName -> HashMap QName v -> Maybe v #-}
 
 newtype Section = Section { _secTelescope :: Telescope }

